src/BusinessLayer/InternetReporting.cpp

```diff
@@ -17,278 +17,355 @@
 #include "MotorDetailsData.h"
 #include "MotorFaultsData.h"
 #include "MpptData.h"
+#include "DataContainer.h"
 
 InternetReporting::InternetReporting(I_CommunicationService& commService,
-                                     const KeyMotorData& keyMotorData,
-                                     const MotorDetailsData& motor0DetailsData,
-                                     const MotorDetailsData& motor1DetailsData,
-                                     const DriverControlsData& driverControlsData,
-                                     const MotorFaultsData& motorFaultsData,
-                                     const BatteryFaultsData& batteryFaultsData,
-                                     const BatteryData& batteryData,
-                                     const MpptData& mpptData,
-                                     const LightsData& lightsData,
-                                     const AuxBmsData& auxBmsData,
+                                     DataContainer& dataContainer0,
+                                     DataContainer& dataContainer1,
                                      InternetView& view)
 
     : communicationService_(commService)
-    , keyMotorData_(keyMotorData)
-    , motor0DetailsData_(motor0DetailsData)
-    , motor1DetailsData_(motor1DetailsData)
-    , driverControlsData_(driverControlsData)
-    , motorFaultsData_(motorFaultsData)
-    , batteryFaultsData_(batteryFaultsData)
-    , batteryData_(batteryData)
-    , mpptData_(mpptData)
-    , lightsData_(lightsData)
-    , auxBmsData_(auxBmsData)
+    , keyMotorData0_(dataContainer0.getKeyMotorData())
```

Comment:
Now that we have the data containers, it would be cleaner to instead make the `internetReporting` class have a vector of datacontainers, and instead of accessing the data member with a private variable, use `dataContainer.getKeyMotorData `